app/scripts/background.js

```diff
@@ -825,37 +827,103 @@ export function setupController(
     updateBadge,
   );
 
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.METAMASK_NOTIFICATIONS_LIST_UPDATED,
+    updateBadge,
+  );
+
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.METAMASK_NOTIFICATIONS_MARK_AS_READ,
+    updateBadge,
+  );
+
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.NOTIFICATIONS_STATE_CHANGE,
+    updateBadge,
+  );
+
   controller.txController.initApprovals();
 
   /**
    * Updates the Web Extension's "badge" number, on the little fox in the toolbar.
    * The number reflects the current number of pending transactions or message signatures needing user approval.
    */
   function updateBadge() {
+    const unapprovedTransactionCount = getUnapprovedTransactionCount();
+    const unreadNotificationsCount = getUnreadNotificationsCount();
+
     let label = '';
-    const count = getUnapprovedTransactionCount();
-    if (count) {
-      label = String(count);
+    // eslint-disable-next-line @metamask/design-tokens/color-no-hex
+    let badgeColor = '#0376C9';
+    if (unapprovedTransactionCount) {
+      label = '\u22EF'; // unicode ellipsis
+    } else if (unreadNotificationsCount > 0) {
+      label =
+        unreadNotificationsCount > 9
+          ? String('9+')
+          : String(unreadNotificationsCount);
+      // eslint-disable-next-line @metamask/design-tokens/color-no-hex
+      badgeColor = '#D73847';
     }
-    // browserAction has been replaced by action in MV3
-    if (isManifestV3) {
-      browser.action.setBadgeText({ text: label });
-      browser.action.setBadgeBackgroundColor({ color: '#037DD6' });
-    } else {
-      browser.browserAction.setBadgeText({ text: label });
-      browser.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
+
+    try {
+      const badgeText = { text: label };
+      const badgeBackgroundColor = { color: badgeColor };
+
+      if (isManifestV3) {
+        browser.action.setBadgeText(badgeText);
+        browser.action.setBadgeBackgroundColor(badgeBackgroundColor);
+      } else {
+        browser.browserAction.setBadgeText(badgeText);
+        browser.browserAction.setBadgeBackgroundColor(badgeBackgroundColor);
+      }
+    } catch (error) {
+      console.error('Error updating browser badge:', error);
     }
```

Comment:
I like the addition of a try-catch (paranoid in case things fail).

Can we also add try-catches down when we compute the counts too? My only reasoning is if the notification count (for whatever reason...) breaks, I don't want it impacting confirmation count.

Reply:
E.g.
```ts
 // if both of these have a try/catch, can default return 0 if they error for whatever reason, so it doesn't impact each-other.
const unapprovedTransactionCount = getUnapprovedTransactionCount();
const unreadNotificationsCount = getUnreadNotificationsCount();
```

Reply:
Logging is important though, we can debug and investigate these through sentry.

Reply:
yep, done @Prithpal-Sooriya !

Reply:
If we're concerned about errors, would it be cleaner to wrap the entire `updateBadge` function in a `try catch` rather than duplicating them in the two get methods?

Reply:
@matthewwalsh0 yes, you're probably right, but I think @Prithpal-Sooriya 's proposal makes sense. By using two try/catch blocks, we can avoid compromising the functionality of one counter if the other fails. What do you think?